src/reactviews/pages/SchemaCompare/components/SchemaDifferences.tsx

```diff
@@ -101,67 +151,163 @@ const SchemaDifferences = ({ onDiffSelected }: Props) => {
             },
         }),
-        createTableColumn<DiffItem>({
+        createTableColumn<DiffEntry>({
             columnId: "targetName",
-            renderHeaderCell: () => loc.schemaCompare.targetName,
             renderCell: (item) => {
                 return <TableCell>{formatName(item.targetValue)}</TableCell>;
             },
         }),
     ];
 
-    const RenderRow = ({ index, style }: any) => {
-        const item = items[index];
+    let items: DiffEntry[] = [];
+    if (compareResult?.success) {
+        items = compareResult.differences.map(
+            (item, index) =>
+                ({
+                    position: index,
+                    ...item,
+                }) as DiffEntry,
+        );
+    }
+
+    const {
+        getRows,
+        selection: { toggleRow },
+    } = useTableFeatures(
+        {
+            columns,
+            items,
+        },
+        [
+            useTableSelection({
+                selectionMode: "multiselect",
+            }),
+        ],
+    );
+
+    const rows: TableRowData[] = getRows((row) => {
+        const selected = row.item.included;
+        return {
+            ...row,
+            onClick: (e: React.MouseEvent) => toggleRow(e, row.rowId),
+            onKeyDown: (e: React.KeyboardEvent) => {
+                if (e.key === " ") {
+                    e.preventDefault();
+                    toggleRow(e, row.rowId);
+                }
+            },
+            selected,
+            appearance: selected ? ("brand" as const) : ("none" as const),
+        };
+    });
+
+    const toggleAllKeydown = (e: React.KeyboardEvent<HTMLDivElement>) => {
+        if (e.key === " ") {
+            handleIncludeExcludeAllNodes();
+            e.preventDefault();
+        }
+    };
+
+    const toggleKeyDown = (
+        e: React.KeyboardEvent<HTMLDivElement>,
+        diffEntry: DiffEntry,
+        include: boolean,
+    ) => {
+        if (e.key === "Enter") {
+            if (diffEntry.position !== undefined) {
+                onDiffSelected(diffEntry.position);
+            }
+            e.preventDefault();
+        }
+        if (e.key === " ") {
+            handleIncludeExcludeNode(diffEntry, include);
+            e.preventDefault();
+        }
+    };
 
+    const RenderRow = ({ index, style, data }: ReactWindowRenderFnProps) => {
+        const { item, appearance, onKeyDown } = data[index];
         return (
             <TableRow
                 aria-rowindex={index + 2}
                 style={style}
-                key={index}
-                // onKeyDown={onKeyDown}
+                key={item.position}
+                onKeyDown={onKeyDown}
                 onClick={() => onDiffSelected(index)}
-                // appearance={appearance}
-            >
-                {columns.map((column) => column.renderCell(item as DiffItem))}
+                appearance={appearance}>
+                <TableCell>{item.name}</TableCell>
```

Comment:
are the `renderCell()` definitions above no longer needed now that you've moved the cell definitions down here?

Reply:
The renderCell definitions are implicitly called on line 307 where I have `{RenderRow}` between the `<List...>{RenderRow}</List>` tags.